cohort-evaluator-spark/src/main/java/com/ibm/cohort/cql/spark/SparkCqlEvaluator.java

```diff
@@ -386,34 +422,36 @@ protected CqlEvaluationRequests readJobSpecification(String path) throws Excepti
      * Write the results of CQL evaluation to a given storage location.
      *
      * @param spark            Active Spark session
+     * @param schema           Schema of the output rows being output.
      * @param resultsByContext CQL evaluation results mapped from context value to a
      *                         map of define to define result.
      * @param outputURI        URI pointing at the location where output data should
      *                         be written.
      */
-    protected void writeResults(SparkSession spark, JavaPairRDD<Object, Map<String, Object>> resultsByContext,
+    protected void writeResults(SparkSession spark, StructType schema, JavaPairRDD<Object, Map<String, Object>> resultsByContext,
             String outputURI) {
-
-//        TODO - output the data in delta lake format.
-//        Map<String,Object> columnData = resultsByContext.take(1).get(0)._2();
-//        StructType schema = new StructType();
-//        schema.add(name, DataTypes.cre)
-//        
-//        JavaRDD<Row> rows = resultsByContext.map( tuple -> {
-//            List<Object> columns = 
-//            
-//            RowFactory.create(tuple._1(), )
-//        })
-        
-        // TODO - use the outputFormat parameter if it isn't null
-        
-
         if (args.outputPartitions != null) {
             resultsByContext = resultsByContext.repartition(args.outputPartitions.intValue());
         }
-        String uuid = UUID.randomUUID().toString();
-        LOG.info("Batch UUID " + uuid);
-        resultsByContext.saveAsTextFile(outputURI + uuid);
+
+        Dataset<Row> dataFrame = spark.createDataFrame(
+                resultsByContext
+                        .map(t -> {
+                            Object contextKey = t._1();
+                            Map<String, Object> results = t._2();
+
+                            List<Object> data = new ArrayList<>();
+                            data.add(contextKey);
+                            for (String field : Arrays.copyOfRange(schema.fieldNames(), 1, schema.fieldNames().length)) {
+                                data.add(results.get(field));
+                            }
+                            return data.toArray();
+                        })
+                        .map(RowFactory::create),
+                schema
+        );
+
+        dataFrame.write().mode(args.overwriteResults ? "overwrite" : "errorifexists").format("parquet").save(outputURI);
```

Comment:
I think it would be good to have an `--output-format` option on the program that allows the user to control the format here.

Reply:
This might be a good item to create a ticket for once we have a use case to write out results using an alternate format.

I know something like delta lake, the most basic write is just swapping out the format string. But I think that there is potentially extra configuration a user would want to be able to provide if writing out results in delta format and I think fully  supporting that would be a nice sized task on its own.

Reply:
My main concern here is that we know our first consumer is going to use delta lake on input and it follows that they will probably expect delta lake on output. I agree that there are other things a user might want to control, but the format itself is easy to add and makes sure delta is available to the end user. As-is, they can't control the output format. I would also be here with simply removing the `.format("parquet")` specifier and allowing the user to use `spark.sql.sources.default` as the control point. That was how I initially did the read portion of the app before adding the `--input-format` option. I think it is best to add `--output-format` though.

Reply:
>My main concern here is that we know our first consumer is going to use delta lake on input and it follows that they will probably expect delta lake on output

I understand this thought, but I think writing out delta format from our program runs counter to the goal of having users write to a particular output location once.  If we write out delta tables, then I feel like that implies that you can re-run the program and append results to a previously created table since that is the main reason you would use delta lake as an output format.

If we want to signal that we only support writing  to separate locations or  overwriting previous results, then not having delta lake as an option probably gets this point across better.

Reply:
I agree with the concern of DeltaLake running counter to our usecase.
No need to take on that extra dressing if we don't actually plan to use it (i.e. persistent results that get updated).
I could possibly see users wanting us to write using other "one-shot" formats like `csv` in the future, but I feel like we could add that later pretty easily.

Reply:
The group consensus appears to be to keep hard-coding parquet for now. I turned my feedback here into a JIRA ticket (WFFHCOHORT-617) and will resolve the conversation so that the PR can move forward.

Reply:
Maybe let's ask HI if they specifically want delta tables as an output from our program. That should make it pretty easy to see if this is something worth adding or not.